cache/test/cache/authentication_test.exs

```diff
@@ -0,0 +1,225 @@
+defmodule Cache.AuthenticationTest do
+  use ExUnit.Case, async: true
+
+  alias Cache.Authentication
+
+  @cache_name :cas_auth_cache
+  @test_auth_header "Bearer test-token-123"
+  @test_server_url "http://localhost:4000"
+
+  setup do
+    Application.put_env(:cache, :cas, server_url: @test_server_url)
+
+    Cachex.clear(@cache_name)
```

Comment:
Same here. I'd scope the cache instance to the test to prevent state leaking across tests. We have some examples of doing this in the server project.